### ISSUE TITLE
How to improve accuracy with partially rotated numbers

### ISSUE THREAD

**Issue by BrunoWouters** [jomjol/AI-on-the-edge-device#317]
First of all thanks for this great project!

The AI seems to struggle with partially rotated number wheels. How can I improve this?
![image](https://user-images.githubusercontent.com/1932784/130152240-f245f25e-191a-481e-a17b-a3ae91039619.png)

Also what is the difference between the inner and outer rectangle when defining the digit references?
I try to keep the number within the inner rectangle with a small space between the edge and the number.
![image](https://user-images.githubusercontent.com/1932784/130152558-20e2f442-8071-4228-bf31-64bf372116d8.png)

**Comment by jomjol** [comment]
Hi @BrunoWouters , 
thanks for you feedback! First of all regarding the two rectangles: it is important, that there is some border around the numbers. Therefore the inner one should fit the number itself like this:
![grafik](https://user-images.githubusercontent.com/30766535/130185574-2cedc34e-0642-4b32-9166-b0acda2faf38.png)

Regarding partially visible numbers, this is a problem in general. You gave an perfect example of this problem:
![grafik](https://user-images.githubusercontent.com/30766535/130185643-c9cb170e-dd0d-40ee-9193-ac0f8b7cff0d.png)
It is even hard for human to distinguish between the 7 and the 2. 

Therefore some consistency checks are implemented, which then will trigger, until the number is correctly recognized.

I can implement your numbers in the next training sets, if you send me 2-3 image for each digit (not more! :-)).

Best regards,
  jomjol

**Comment by Sunfllower** [comment]
Is it not possible for the most right digit to detect if the number goes up or down? Once there is no horizontal black space in de area where to number should be a number must be readable. Also keep in mind the last number (for example 7) then the next number should be either a 6 or a 8.

**Comment by jomjol** [comment]
@Sunfllower : This is not possible with the current usage of a classification network. That would be a major change in the neural network. Really not trivial, already tested in some first attempt.

**Comment by krikk** [comment]
would it be possible to implement an option to only "accept" values where all digits are correctly detected, so to ignore values where we have a "N" in it, one example why this would be nice:

![image](https://user-images.githubusercontent.com/5165445/131227809-a25da6f0-4432-4929-9423-faec41faee8f.png)
 old detected value:   {"value":167.9492,"error":"no error","rate":0.000666,"timestamp":"2021-08-28T20:11:19"}
--> CORRECT !
 
![image](https://user-images.githubusercontent.com/5165445/131227826-8dfacef6-311a-4d83-8297-e943ae61d964.png)
 new detected value:  {"value":167.9592,"error":"no error","rate":0.003337,"timestamp":"2021-08-28T20:14:19"}
WRONG! ..i know how this happens, it can not read the last digit, so it takes the pre-value -> wrong value which also causes problems on the next values because of "Neg. Rate - Read" on follow up values...

or are there any other options to correct this problem, which happens quite frequently on both of my water meters...

**Comment by jomjol** [comment]
@krikk : you can disable to use the prevalue. In this case the "N" will not be replaced and transmitted. In your own postprocessing in the MQTT broker you can then dismiss the values if they have a "N" and do not send them to the database.

**Comment by pixeldoc2000** [comment]
![grafik](https://user-images.githubusercontent.com/376715/131917241-23d2c301-0041-48e6-971d-4d431c4a137b.png)
Simular issue: This is Recognised as 72928 or 72935 before.

Reference Image:
![grafik](https://user-images.githubusercontent.com/376715/131917826-aac30929-23f2-4687-8e58-5c3614409273.png)
Is the Border ok?

**Comment by jomjol** [comment]
@pixeldoc2000 Make the ROIs smaller, the inner rectangle is tought to be exactly around the number without any space

**Comment by jomjol** [comment]
I'll close this for the time beeing. Please reopen in case of need.
